server/db/db.js

```diff
@@ -24,7 +24,11 @@ pool.on('error', err => {
 types.setTypeParser(types.builtins.NUMERIC, value => parseFloat(value))
 
 function createSessionFromRow(r, allButtons) {
-  const button = allButtons.filter(b => b.id === r.button_id)[0]
+  let button // initially undefined
+
+  if (allButtons !== undefined) {
+    button = allButtons.filter(b => b.id === r.button_id)[0]
+  }
```

Comment:
We talked about this earlier, so I'll just note down my understanding of where we got to. 

Good options are either:
* Filter out the button when you want to in the api.js functions, or
* Create new db.* functions with names that make it clear that they will not contain the full button

Reply:
I've decided to keep the DB functions as they are, with full client objects, then filter these from within api.js.